File: backend-api-dot-net/KPW.Domain/Entities/Exercise.cs
using KPW.Domain.Common;

namespace KPW.Domain.Entities;

public class Exercise : AuditableEntity
{
    public int ExerciseId { get; set; }
    public string Title { get; set; } = string.Empty;
    public string? ShortDescription { get; set; }
    public string? TargetedMuscles { get; set; }
    public string? ClinicalPurpose { get; set; }
    public string? SafetyNotes { get; set; }
    public string? CommonMistakes { get; set; }
    public string? VideoUrl { get; set; }
    public string? TargetSpecies { get; set; }
    public string? ConditionCategory { get; set; }
    public int DifficultyLevel { get; set; } = 1;
    public string? CoverImageUrl { get; set; }

    public bool IsSystemDefault { get; set; } = true;
    public int? ClinicId { get; set; }
    public int? BaseExerciseId { get; set; }
    public bool IsActiveForOwners { get; set; } = true;

    public Clinic? Clinic { get; set; }
    public Exercise? BaseExercise { get; set; }
    public ICollection<Exercise> CustomOverrides { get; set; } = [];

    public ICollection<ExerciseStep> Steps { get; set; } = [];
    public ICollection<RehabProgramExercise> RehabProgramExercises { get; set; } = [];
    public ICollection<VideoSubmission> VideoSubmissions { get; set; } = [];
}

File: physio-portal-vue3/src/types/exercise.ts
export interface ExerciseStep {
  exerciseStepId: number
  stepNumber: number
  stepInstruction: string
  imageUrl: string | null
}

export interface Exercise {
  exerciseId: number
  title: string
  shortDescription: string | null
  targetedMuscles: string | null
  clinicalPurpose: string | null
  safetyNotes: string | null
  commonMistakes: string | null
  videoUrl: string | null
  targetSpecies: string | null
  conditionCategory: string | null
  difficultyLevel: number
  steps: ExerciseStep[]
}

export interface CreateExerciseStepRequest {
  stepNumber: number
  stepInstruction: string
  imageUrl?: string
}

export interface CreateExerciseRequest {
  title: string
  shortDescription?: string
  targetedMuscles?: string
  clinicalPurpose?: string
  safetyNotes?: string
  commonMistakes?: string
  videoUrl?: string
  targetSpecies?: string
  conditionCategory?: string
  difficultyLevel: number
  steps?: CreateExerciseStepRequest[]
}

export interface ProgramBuilderExercise {
  exerciseId: number
  title: string
  shortDescription: string | null
  repetitions: number
  sets: number
  frequencyPerDay: number
}

export interface CreateRehabProgramExercise {
  exerciseId: number
  repetitions: number
  sets: number
  frequencyPerDay: number
  phaseId?: number
}

export interface CreateRehabProgramRequest {
  petId: number
  programTitle: string
  startDate: string
  endDate?: string
  notes?: string
  exercises: CreateRehabProgramExercise[]
}

export interface RehabProgramExercise {
  rehabProgramExerciseId: number
  exerciseId: number
  title: string
  repetitions: number
  sets: number
  frequencyPerDay: number
  phaseId?: number
  shortDescription: string | null
  safetyNotes: string | null
  commonMistakes: string | null
  videoUrl: string | null
  steps: ExerciseStep[]
}

export interface RehabProgram {
  rehabProgramId: number
  physioId: number
  petId: number
  programTitle: string
  startDate: string
  endDate: string | null
  notes: string | null
  exercises: RehabProgramExercise[]
}
